sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/receivesettle/async/OrderAsyncService.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus.receivesettle.async;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.receivesettle.NetworkFailureException;
+import com.azure.messaging.servicebus.receivesettle.Order;
+import com.azure.messaging.servicebus.receivesettle.OrderServiceFailureException;
+import com.azure.messaging.servicebus.receivesettle.sync.OrderSyncService;
+import reactor.core.publisher.Mono;
+
+import java.util.Random;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+/**
+ * An order service that save orders and possibly throw exception for each operation.
+ */
+public class OrderAsyncService {
+    private final Random rand = new Random();
+    private final ClientLogger logger = new ClientLogger(OrderSyncService.class);
+
+    /**
+     * Simulate an order creation or update
+     * Exception might be thrown before or after the order is saved (logged)
+     * @param order The order to create or replace.
+     * @return Mono
+     */
+    public Mono<Void> createOrReplaceOrder(Order order) {
+        return this.throwRandomError()
+            .then(Mono.fromRunnable(() -> {
+                if (order.getId() == null) {
+                    Order newOrder = new Order();
+                    newOrder.setId(UUID.randomUUID().toString());
+                    newOrder.setContent(order.getContent());
+                    // simulate creating a new order
+                    logger.info(String.format("Order created: %s", newOrder));
+                } else {
+                    // simulate updating the order into repository.
+                    logger.info(String.format("Order updated %s", order));
+                }
+            })).then(this.throwRandomError());
```

Comment:
Why do we want to throw a random error before and after the order creation?

Reply:
In reality, you might have exceptions thrown before or after the data are successfully saved.
I can remove the 2nd one if you think it's not necessary.